### PR TITLE
fix inbox from briefly displaying when creating WS from selector and …

### PR DESCRIPTION
**Pull request by huult**
### Fixed Issues
$ https://github.com/Expensify/App/issues/50850
PROPOSAL: https://github.com/Expensify/App/issues/50850#issuecomment-2440894414

### Tests

Same QA step

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open the Expensify app.
2. Tap the workspace selector icon on the top left.
3. Tap the "+" icon to add a new workspace.
4. Complete any required steps to create the workspace.
5. Verify navigation to the new workspace menu after creation.
6. Tap the back arrow in the new workspace menu.
7. Confirm redirection to the workspace selector screen.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/98d68a88-e5f3-4757-a09f-f74950f74ae3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/f556717a-dd62-4a4e-b510-b0552a47734e

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/fabe7c25-e45b-455a-8f4c-2b162407be06

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/71176ef1-960d-4863-a62a-193cc3fd4474

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/d20ff6ac-23f7-4ae6-9353-4c59522ebe04

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/052a4c14-eeb0-40a4-aff7-f1cd103293ca

</details>

### CODE REVIEW

**Comment by huult** [comment]
Hi @dominictb, @puneetlath We have an issue after updating the pull request for this ticket, which is:

- The LHNOptionsList sometimes does not render the full list of data (You can watch the video in this pull request to see the issue I mentioned) after reset to home. This is a separate RCA issue for this ticket and is currently present in production, staging, and the main branch.

Can I ask if we are fixing this issue in this pull request or if a new ticket will be created to address it?

**Comment by puneetlath** [comment]
Interesting? What's the reason for that bug?

**Comment by puneetlath** [comment]
Ok, can you look into it? If it's a straightforward change, we can include it hear (and increase the pay). If it's more involved, we can handle separately.

**Comment by huult** [comment]
@puneetlath , @dominictb I've updated the fix for the issue mentioned above.

The RCA:

When we tap the plus icon, the [orderedReportIDs](https://github.com/Expensify/App/blob/8d3f1db0ef1b28c5ef9c03f251d9b9dc2adaed4e/src/pages/home/sidebar/SidebarLinksData.tsx#L40) are updated with the new workspace we created. At that moment, the [Flashlist](https://github.com/Expensify/App/blob/9d252e8b37c441d62d6b299421870d28fe2ed398/src/components/LHNOptionsList/LHNOptionsList.tsx#L267) also updates because the list data has changed.

However, we then navigate to the workspace initial screen, so FlashList unmounts, and the [LayoutManager](node_modules/recyclerlistview/src/core/layoutmanager/LayoutManager.ts) inside FlashList sets the height to 0. When we return from the workspace, the data hasn’t changed because it was updated previously, so the height remains 0, and this causes the issue.

This issue happens on mWeb and web. On web, the behavior can appear random, as it depends on the workspace created. If the created workspace includes an admin room in the LHN, the issue doesn’t occur because there is new data, causing FlashList to re-render.

The solutions
We should delay navigation until FlashList has finished updating.

If you have any questions, let me know. I can explain further, as I have a deep understanding of why this happened.

**Comment by huult** [comment]
@puneetlath @dominictb Can you check my comment?

**Comment by huult** [comment]
@dominictb 
> The bug you are talking about is: LHN just show only 1 item?

Yes

> I can't open the LayoutManager reference link.

`node_modules/recyclerlistview/dist/reactnative/core/layoutmanager/LayoutManager.js` here is the path LayoutManager

> If the FlashList unmounts, how the component inside it sets the height to 0 (since it unmounts too)

I will provide you with more details tomorrow.
This video shows the height to 0 that I mentioned

https://github.com/user-attachments/assets/49820d77-14b3-4b9b-874f-6a33b20705c7

**Comment by huult** [comment]
@dominictb I am busy at this time. I will give it as soon as possible, but no later than tomorrow.

**Comment by huult** [comment]
@dominictb 

> If the FlashList unmounts, how the component inside it sets the height to 0 (since it unmounts too)

When tapping the plus icon, the data in FlashList changes, triggering the componentDidUpdate function in node_modules/recyclerlistview/dist/reactnative/core/RecyclerListView.js. Within this function, _checkAndChangeLayouts recalculates the layout (using relayoutFromIndex) when the data changes. That is the logic of FlashList.

https://github.com/user-attachments/assets/99ef0fa5-2437-41a7-859f-f7bba0416937

But when `relayoutFromIndex` is executed, and we navigate to the workspace, `relayoutFromIndex` returns a height of 0. We able debug node_modules/recyclerlistview/dist/reactnative/core/layoutmanager/LayoutManager.js to see it. That why, in `div` set to 0 in this video.

https://github.com/user-attachments/assets/f2b4272c-947a-44b1-b4de-a9503e48550f

And when we go back to the home screen, the height value is still 0. Since the data hasn’t changed, componentDidUpdate is not triggered to re-layout, and this issue occurs.

My suggested solution is to delay navigation to the workspace until the FlashList re-layout is complete. That’s why I used 
```diff
//src/libs/actions/App.ts#L380
      InteractionManager.runAfterInteractions(() => {
                Navigation.navigate(ROUTES.WORKSPACE_INITIAL.getRoute(policyID, backTo));
            });
```

**Comment by huult** [comment]
@dominictb @puneetlath https://github.com/Expensify/App/issues/52651  is the same RCA we know, and I have found the RCA and have a fix.

**Comment by huult** [comment]
@dominictb You are correct, if we removed the plus icon, this issue would no longer be valid, and the issue I mentioned would still not be reproducible

**Comment by huult** [comment]
@dominictb When we remove the plus icon, we still encounter the 'get started' case for when there is no workspace. So I think we are able to proceed with this pull request to fix this case.

This issue I mentioned:

https://github.com/user-attachments/assets/8c2c45b3-096f-4608-94dd-a1d11190a5bc

I have already updated the pull request to fix the issue with the 'Get started' case. Can you check it again? Thank you!

File: src/libs/actions/App.ts [hunk 1]
```diff
@@ -378,7 +378,9 @@ function createWorkspaceWithPolicyDraftAndNavigateToIt(policyOwnerEmail = '', po
                 Navigation.goBack();
             }
             savePolicyDraftByNewWorkspace(policyID, policyName, policyOwnerEmail, makeMeAdmin);
-            Navigation.navigate(ROUTES.WORKSPACE_INITIAL.getRoute(policyID, backTo));
+            InteractionManager.runAfterInteractions(() => {
```
[the remark this answers is not in the document]
Comment (huult) [reply src/libs/actions/App.ts:381]: yes

**Comment by puneetlath** [comment]
Hi! Just curious for a summary on where we're at with this PR.

**Comment by huult** [comment]
As discussed in the ticket, I will close this pull request

**Pull request closed without merging**